### ISSUE TITLE
AdvancedEnchantments Explosive Enchant

### ISSUE THREAD

**Issue by Skizzles** [GriefPrevention/GriefPrevention#1771]
### Observed Behavior

Untrusted players can remove item frames in a claim using AE's Explosive enchant.

### Expected Behavior

To block the explosive damage just like standard explosions.

### Reproduction steps

Took a bow with explosive
Shot an entity (zombie) near item frames in a claim
Watched as item frames fell off the wall

### Stack trace or error log

```shell
N/A
```

### Server version

```shell
This server is running Purpur version git-Purpur-1467 (MC: 1.18.1) (Implementing API version 1.18.1-R0.1-SNAPSHOT) (Git: 288d42c on HEAD)
You are 61 version(s) behind
Download the new version at: https://purpurmc.org/downloads
Previous version: git-Purpur-1477 (MC: 1.18.1)
```

### GriefPrevention version

```shell
GriefPrevention version 16.17.1
```

### Configuration

```yaml
# Default values are perfect for most servers.  If you want to customize and have a question, look for the answer here first: http://dev.bukkit.org/bukkit-plugins/grief-prevention/pages/setup-and-configuration/
GriefPrevention:
  SeaLevelOverrides:
    Spawn_World: -1
    Spawn_World_nether: -1
    Spawn_World_the_end: -1
    Resource_World: -1
    Build_World: -1
    Build_World_nether: -1
    Build_World_the_end: -1
  Claims:
    Mode:
      Spawn_World_the_end: Disabled
      Spawn_World_nether: Disabled
      Resource_World: Disabled
      Build_World: Survival
      Build_World_the_end: Survival
      Spawn_World: Disabled
      Build_World_nether: Survival
    PreventGlobalMonsterEggs: true
    PreventTheft: true
    ProtectCreatures: true
    PreventButtonsSwitches: true
    LockWoodenDoors: false
    LockTrapDoors: false
    LockFenceGates: true
    EnderPearlsRequireAccessTrust: true
    RaidTriggersRequireBuildTrust: true
    ProtectHorses: true
    ProtectDonkeys: true
    ProtectLlamas: true
    InitialBlocks: 400
    Claim Blocks Accrued Per Hour:
      Default: 200
    Max Accrued Claim Blocks:
      Default: 80000
    Accrued Idle Threshold: 0
    AccruedIdlePercent: 0
    AbandonReturnRatio: 1.0
    AutomaticNewPlayerClaimsRadius: 4
    AutomaticNewPlayerClaimsRadiusMinimum: 0
    ExtendIntoGroundDistance: 255
    MinimumWidth: 5
    MinimumArea: 100
    MaximumDepth: 0
    InvestigationTool: STICK
    ModificationTool: FLINT
    Expiration:
      ChestClaimDays: 30
      UnusedClaimDays: 30
      AllClaims:
        DaysInactive: 120
        ExceptWhenOwnerHasTotalClaimBlocks: 1000000
        ExceptWhenOwnerHasBonusClaimBlocks: 500000
      AutomaticNatureRestoration:
        SurvivalWorlds: true
    AllowTrappedInAdminClaims: false
    MaximumNumberOfClaimsPerPlayer: 0
    CreationRequiresWorldGuardBuildPermission: true
    VillagerTradingRequiresPermission: true
    CommandsRequiringAccessTrust: /sethome;/esethome;/spawner;/pwarp create
    DeliverManuals: true
    ManualDeliveryDelaySeconds: 30
    RavagersBreakBlocks: true
    FireSpreadsInClaims: false
    FireDamagesInClaims: false
    LecternReadingRequiresAccessTrust: true
  Spam:
    Enabled: false
    LoginCooldownSeconds: 60
    LoginLogoutNotificationsPerMinute: 5
    ChatSlashCommands: /me;/global;/local
    WhisperSlashCommands: /tell;/pm;/r;/whisper;/msg
    WarningMessage: Please reduce your noise level.  Spammers will be banned.
    BanOffenders: true
    BanMessage: Banned for spam.
    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
    DeathMessageCooldownSeconds: 120
    Logout Message Delay In Seconds: 0
  PvP:
    RulesEnabledInWorld:
      Spawn_World: false
      Spawn_World_nether: true
      Spawn_World_the_end: true
      Resource_World: true
      Build_World: true
      Build_World_nether: true
      Build_World_the_end: true
    ProtectFreshSpawns: false
    PunishLogout: false
    CombatTimeoutSeconds: 15
    AllowCombatItemDrop: true
    BlockedSlashCommands: /home;/vanish;/spawn;/tpa;/pwarp;/pw;/fly
    ProtectPlayersInLandClaims:
      PlayerOwnedClaims: true
      AdministrativeClaims: true
      AdministrativeSubdivisions: true
    AllowLavaDumpingNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    AllowFlintAndSteelNearOtherPlayers:
      PvPWorlds: true
      NonPvPWorlds: false
    ProtectPetsOutsideLandClaims: true
  Economy:
    ClaimBlocksMaxBonus: 80000
    ClaimBlocksPurchaseCost: 250.0
    ClaimBlocksSellValue: 50.0
  ProtectItemsDroppedOnDeath:
    PvPWorlds: false
    NonPvPWorlds: false
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: false
  BlockSurfaceOtherExplosions: false
  LimitSkyTrees: true
  LimitTreeGrowth: false
  PistonMovement: EVERYWHERE_SIMPLE
  PistonExplosionSound: true
  FireSpreads: true
  FireDestroys: true
  AdminsGetWhispers: false
  AdminsGetSignNotifications: true
  SmartBan: true
  Mute New Players Using Banned Words: false
  MaxPlayersPerIpAddress: 3
  SilenceBans: true
  Siege:
    Worlds: []
    BreakableBlocks:
    - DIRT
    - GRASS_BLOCK
    - GRASS
    - FERN
    - DEAD_BUSH
    - COBBLESTONE
    - GRAVEL
    - SAND
    - GLASS
    - GLASS_PANE
    - OAK_PLANKS
    - SPRUCE_PLANKS
    - BIRCH_PLANKS
    - JUNGLE_PLANKS
    - ACACIA_PLANKS
    - DARK_OAK_PLANKS
    - WHITE_WOOL
    - ORANGE_WOOL
    - MAGENTA_WOOL
    - LIGHT_BLUE_WOOL
    - YELLOW_WOOL
    - LIME_WOOL
    - PINK_WOOL
    - GRAY_WOOL
    - LIGHT_GRAY_WOOL
    - CYAN_WOOL
    - PURPLE_WOOL
    - BLUE_WOOL
    - BROWN_WOOL
    - GREEN_WOOL
    - RED_WOOL
    - BLACK_WOOL
    - SNOW
    DoorsOpenDelayInSeconds: 300
    CooldownEndInMinutes: 60
  EndermenMoveBlocks: false
  SilverfishBreakBlocks: false
  CreaturesTrampleCrops: false
  RabbitsEatCrops: true
  HardModeZombiesBreakDoors: false
  Database:
    URL: ''
    UserName: ''
    Password: ''
  UseBanCommand: false
  BanCommandPattern: ban %name% %reason%
  Advanced:
    fixNegativeClaimblockAmounts: true
    ClaimExpirationCheckRate: 60
    OfflinePlayer_cache_days: 120
  Abridged Logs:
    Days To Keep: 7
    Included Entry Types:
      Social Activity: true
      Suspicious Activity: true
      Administrative Activity: false
      Debug: false
      Muted Chat Messages: false
```

### Plugin list

```shell
AdvancedEnchantments
GriefPrevention
```

### Running without GriefPrevention

- [ ] I attempted running the server without GriefPrevention installed.
- [X] The problem does _not_ occur when GriefPrevention is removed from the server.

### Running with only GriefPrevention

- [ ] I attempted running only GriefPrevention on the server.
- [X] The issue still occurs when GriefPrevention is the only plugin running.

### Running on a fresh, clean server installation

- [ ] I attempted testing for the issue on a new server.
- [X] The issue still occurs on a new server.

### Using unmodified client

- [ ] I attempted testing for the issue with the vanilla client.
- [X] The issue still occurs when using the vanilla client.

### We appreciate you taking the time to fill out a bug report!

- [X] I searched for similar issues before submitting this bug report.

**Comment by bigpresh** [comment]
If the explosions triggered by AdvancedEnchantments aren't standard Bukkit events, or AE doesn't check if the events have been cancelled, then it won't work.  GP can't know about every possible third party plugin out there, but if they trigger suitable Bukkit events it should work well.

Someone would probably need to reproduce the problem with event debugging to see exactly which events are fired.

**Comment by Jikoo** [comment]
Going to close this as AE is a closed-source premium plugin - if you want assistance with compatibility, you will need to contact its developer. There's nothing we can do without more information that we can't obtain because we don't have access to the plugin.

If you want help here, run EventDebug and get us logs from when the enchantment is used.

**Comment by Burchard36** [comment]
The issue is fixed as of version 8.9.11 for AdvancedEnchantments: https://github.com/GC-spigot/AdvancedEnchantments/issues/2743
